uber/config.py

```diff
@@ -197,103 +253,95 @@ def _is_intstr(s):
-EVENT_MONTH = EPOCH.strftime('%B')
-EVENT_START_DAY = int(EPOCH.strftime('%d')) % 100
-EVENT_END_DAY = int(ESCHATON.strftime('%d')) % 100
+c.CON_LENGTH = int((c.ESCHATON - c.EPOCH).total_seconds() // 3600)
+c.START_TIME_OPTS = [(dt, dt.strftime('%I %p %a')) for dt in (c.EPOCH + timedelta(hours=i) for i in range(c.CON_LENGTH))]
+c.DURATION_OPTS = [(i, '%i hour%s' % (i, ('s' if i > 1 else ''))) for i in range(1, 9)]
+c.EVENT_START_TIME_OPTS = [(dt, dt.strftime('%I %p %a') if not dt.minute else dt.strftime('%I:%M %a'))
+                           for dt in [c.EPOCH + timedelta(minutes=i * 30) for i in range(2 * c.CON_LENGTH)]]
+c.EVENT_DURATION_OPTS = [(i, '%.1f hour%s' % (i/2, 's' if i != 2 else '')) for i in range(1, 19)]
+c.SETUP_TIME_OPTS = [(dt, dt.strftime('%I %p %a')) for dt in (c.EPOCH - timedelta(days=2) + timedelta(hours=i) for i in range(16))] \
+                  + [(dt, dt.strftime('%I %p %a')) for dt in (c.EPOCH - timedelta(days=1) + timedelta(hours=i) for i in range(24))]
+c.TEARDOWN_TIME_OPTS = [(dt, dt.strftime('%I %p %a')) for dt in (c.ESCHATON + timedelta(hours=i) for i in range(6))] \
+                     + [(dt, dt.strftime('%I %p %a'))
+                        for dt in ((c.ESCHATON + timedelta(days=1)).replace(hour=10) + timedelta(hours=i) for i in range(12))]
 
-DAYS = sorted({(dt.strftime('%Y-%m-%d'), dt.strftime('%a')) for dt,desc in START_TIME_OPTS})
-HOURS = ['{:02}'.format(i) for i in range(24)]
-MINUTES = ['{:02}'.format(i) for i in range(60)]
 
-ORDERED_EVENT_LOCS = [loc for loc, desc in EVENT_LOCATION_OPTS]
-EVENT_BOOKED = {'colspan': 0}
-EVENT_OPEN   = {'colspan': 1}
+c.EVENT_NAME_AND_YEAR = c.EVENT_NAME + (' {}'.format(c.YEAR) if c.YEAR else '')
+c.EVENT_YEAR = c.EPOCH.strftime('%Y')
+c.EVENT_MONTH = c.EPOCH.strftime('%B')
+c.EVENT_START_DAY = int(c.EPOCH.strftime('%d')) % 100
+c.EVENT_END_DAY = int(c.ESCHATON.strftime('%d')) % 100
 
-MAX_BADGE = max(xs[1] for xs in BADGE_RANGES.values())
+c.DAYS = sorted({(dt.strftime('%Y-%m-%d'), dt.strftime('%a')) for dt, desc in c.START_TIME_OPTS})
+c.HOURS = ['{:02}'.format(i) for i in range(24)]
+c.MINUTES = ['{:02}'.format(i) for i in range(60)]
 
-JOB_PAGE_OPTS = (
+c.ORDERED_EVENT_LOCS = [loc for loc, desc in c.EVENT_LOCATION_OPTS]
+c.EVENT_BOOKED = {'colspan': 0}
+c.EVENT_OPEN   = {'colspan': 1}
+
+c.MAX_BADGE = max(xs[1] for xs in c.BADGE_RANGES.values())
+
+c.JOB_PAGE_OPTS = (
     ('index',    'Calendar View'),
     ('signups',  'Signups View'),
     ('staffers', 'Staffer Summary')
 )
-WEIGHT_OPTS = (
+c.WEIGHT_OPTS = (
     ('1.0', 'x1.0'),
     ('1.5', 'x1.5'),
     ('2.0', 'x2.0'),
     ('2.5', 'x2.5'),
 )
-JOB_DEFAULTS = ['name', 'description', 'duration', 'slots', 'weight', 'restricted', 'extra15']
-
-TABLE_OPTS = [
-    (0.5, 'Half Table'),
-    (1.0, 'Full Table'),
-    (2.0, 'Double Table'),
-    (3.0, 'Triple Table'),
-    (4.0, 'Island/Quad Table')
-]
-
-ADMIN_TABLE_OPTS = [(0.0, 'No Table')] + TABLE_OPTS + [(float(i), str(i)) for i in range(5, 11)]
```

Comment:
Looking at it again, `ADMIN_TABLE_OPTS` takes the regular options (which, granted, were hard-coded) and adds 0 to it plus some extra numbers.

The `range(5,11)` should be changed to something that simply counts up +5 or thereabouts, but otherwise it looks like a candidate for keeping in the main repo.